projects/packages/videopress/src/class-wpcom-rest-api-v2-attachment-field-videopress.php

```diff
@@ -91,15 +93,12 @@ public function get_schema() {
 	 * @return string
 	 */
 	public function get( $attachment, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-			$blog_id = get_current_blog_id();
-		} else {
-			$blog_id = \Jetpack_Options::get_option( 'id' );
+		if ( ! isset( $attachment['id'] ) ) {
+			return array();
 		}
 
-		$post_id = absint( $attachment['id'] );
-
-		$videopress_guid = $this->get_videopress_guid( $post_id, $blog_id );
+		$blog_id         = Jetpack_Connection::get_site_id();
```

Comment:
I took this opportunity to simplify how we get the blog ID here, since we already have an extracted method to do just that in the Connection package.

Reply:
Do you think we should check for `WP_Error` here to properly handle connection errors?
https://github.com/Automattic/jetpack/blob/f00a638a9b881f9c648b70df94a0bf9006b49de7/projects/packages/connection/src/class-manager.php#L2572

Reply:
That makes sense. What do you think of 3b790e5ab0223c6fcce73d80c96bfdee3e704367?

Reply:
That should work, thanks for taking care of it!